### PR TITLE
feat: Adapt Chip to respect Spark specs (#343)

### LINKED ISSUE #343: [Components] Chip integration

### PR DESCRIPTION
**Pull request by MarinaRomanova**
## 📋 Changes description
- Update Chip component to respect Spark specs.

## 📸 Screenshots
Generated by paparazzi in this PR files

## 🗒️ Other info
close #343

### CODE REVIEW

**Review by shkschneider — APPROVED** [review]

File: spark/src/main/kotlin/com/adevinta/spark/components/chips/AssistChip.kt [hunk 1]
```diff
@@ -93,6 +93,14 @@ import androidx.compose.material3.ElevatedAssistChip as MaterialElevatedAssistCh
  * for this chip. You can create and pass in your own `remember`ed instance to observe
  * [Interaction]s and customize the appearance / behavior of this chip in different states.
  */
+@Deprecated(
+    message = "Use one of the options: ChipOutlined, ChipFilled, ChipTinted, ChipDashed",
```
Comment (soulcramer) [inline spark/src/main/kotlin/com/adevinta/spark/components/chips/AssistChip.kt:97]: Material Assyst Chip is closer to Outlined than the other styles variant so we should recommend this one only instead

File: spark/src/main/kotlin/com/adevinta/spark/components/chips/FilterChip.kt [hunk 2]
```diff
@@ -105,6 +90,10 @@ import androidx.compose.material3.FilterChip as MaterialFilterChip
  * for this chip. You can create and pass in your own `remember`ed instance to observe
  * [Interaction]s and customize the appearance / behavior of this chip in different states.
  */
+@Deprecated(
+    "Use one of the options: ChipOutlined, ChipFilled, ChipTinted, ChipDashed",
```
Comment (soulcramer) [inline spark/src/main/kotlin/com/adevinta/spark/components/chips/FilterChip.kt:94]: Material Filter Chip is closer to TintedChip than the other styles variant so we should recommend this one only instead

File: spark/src/main/kotlin/com/adevinta/spark/components/chips/InputChip.kt [hunk 3]
```diff
@@ -111,6 +90,14 @@ import androidx.compose.material3.InputChip as MaterialInputChip
  * for this chip. You can create and pass in your own `remember`ed instance to observe
  * [Interaction]s and customize the appearance / behavior of this chip in different states.
  */
+@Deprecated(
+    message = "Use one of the options: ChipOutlined, ChipFilled, ChipTinted, ChipDashed",
```
Comment (soulcramer) [inline spark/src/main/kotlin/com/adevinta/spark/components/chips/InputChip.kt:94]: Material Input Chip is closer to Outlined than the other styles variant so we should recommend this one only instead

File: spark/src/main/kotlin/com/adevinta/spark/components/chips/SuggestionChip.kt [hunk 4]
```diff
@@ -87,6 +79,14 @@ import androidx.compose.material3.SuggestionChip as MaterialSuggestionChip
  * for this chip. You can create and pass in your own `remember`ed instance to observe
  * [Interaction]s and customize the appearance / behavior of this chip in different states.
  */
+@Deprecated(
+    message = "Use one of the options: ChipOutlined, ChipFilled, ChipTinted, ChipDashed",
```
Comment (soulcramer) [inline spark/src/main/kotlin/com/adevinta/spark/components/chips/SuggestionChip.kt:83]: Material Suggestion Chip is closer to Outlined than the other styles variant so we should recommend this one only instead

**Pull request merged**